spec/readme.md

```diff
@@ -0,0 +1,102 @@
+# Overview
+
+# Definitions
+* [Labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/) are key/value pairs that are attached to objects. Labels are intended to be used to specify identifying attributes of objects that are meaningful and relevant to users, but do not directly imply semantics to the core system. Labels can be used to organize and to select subsets of objects.
+* [Annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/) are arbitrary non-identifying metadata attached to objects.
+* Artifact
+* Component is the manifestation of deployed artifacts. A component can only be manifested by one, and only one, artifact.
+* Server
+* Interface
+
+# Workflow
+1. POST _Artifact Specification_ to `/deployments/start` which returns the _Deployment Specification Plan_ with a `plan id`
+  * process all StringTemplate
+  * Check if a `System` node exists matching `system` name
+  * if `Artifact` node matching the `name` does not exist, create a `Artifact` node for the `System`
+  * if `ArtifactVersion` node of type `name` does not exist, create a `ArtifactVersion` node for the `System`
+  * Create `DeploymentSpec` node for the `System`, `ArtifactVersion` and `environment`
+  * Process `runtimeSpec.provides`
+    * if `Interface` node matching `interface` does not exist, creates a `Interface` node for the `System`
+    * Create of Update a `Component` node for the `DeploymentSpec` node withe `Interface` node as the `interface` attribute
+  * Process `runtimeSpec.requires`
+  * Process `deploymentSpec`
+1. Parse _Deployment Specification Plan_, extract parameters/arguments and execute deployment.
+1. POST the _Deployment Specification Plan_ to `/deployments/<plan id>/finish` indicating if it was successful or not.
+
```

Comment:
Do we need to post the deployment specification plan again?  If so, would the finish update the data store if the object has changed?

Reply:
@GeorgeWalker, I was thinking the deployment process could provide some extra metadata (mostly attributes). But I am good with defining (or at least starting with) as one-way.